### ISSUE TITLE
Catch failures in add_node

### ISSUE THREAD

**Issue by roydahan** [scylladb/scylla-cluster-tests#754]
In case there is a failure to add new node (e.g after decommission or terminate_and_replace), we should catch it and:
1. Remove it from cluster.db_nodes list.
2. Retry adding node.
(We can set max number of retries to 3).

**Comment by roydahan** [comment]
```
2019-01-19 05:33:56,228 nemesis          L0448 ERROR| sdcm.nemesis.ChaosMonkey: Exception in random_disrupt_method terminate_and_replace_node: Setup has failed!
2019-01-19 05:33:56,229 nemesis          L0734 ERROR| sdcm.nemesis.ChaosMonkey: Unhandled exception in method <function disrupt at 0x7fc248b01050>
Traceback (most recent call last):
  File "/sct/sdcm/nemesis.py", line 884, in disrupt
  File "/sct/sdcm/nemesis.py", line 446, in call_random_disrupt_method
  File "/sct/sdcm/nemesis.py", line 303, in disrupt_terminate_and_replace_node
  File "/sct/sdcm/nemesis.py", line 248, in _add_and_init_new_cluster_node
  File "/sct/sdcm/cluster.py", line 1386, in wrapper
AssertionError: Setup has failed!
```

**Comment by roydahan** [comment]
At the moment no need to destroy the nodes we created but failed to setup, so we will be able to investigate later why the setup has failed.

**Comment by roydahan** [comment]
If possible to quickly fix it, let's backport to branch-3.0.

**Comment by roydahan** [comment]
I meant to retry the entire _add_and_init..

Can you please queue it with high priority?
